src/foreground-plugin.ts

```diff
@@ -1,23 +1,40 @@
 import { BEGIN, CallContext, END, ENV, GET_BLOCK, RESET, SET_HOST_CONTEXT, STORE } from './call-context.ts';
 import { type InternalConfig, InternalWasi, PluginOutput } from './interfaces.ts';
+import { CAPABILITIES } from './polyfills/deno-capabilities.ts';
 import { loadWasi } from './polyfills/deno-wasi.ts';
 
 export const EXTISM_ENV = 'extism:host/env';
 
 type InstantiatedModule = [WebAssembly.Module, WebAssembly.Instance];
 
+interface SuspendingCtor {
+  new(fn: CallableFunction): any
+}
+
+const AsyncFunction = (async () => { }).constructor
```

Comment:
This is when I realized I knew nothing about JS. Usually it happens a little further down the diff :sweat_smile: